lib/fika/compiler/code_server/function_dependencies.ex

```diff
@@ -0,0 +1,34 @@
+defmodule Fika.Compiler.CodeServer.FunctionDependencies do
```

Comment:
@emilsoman I ended up keeping this as a separate submodule because in this way we can use the same graph logic for handling loops in the sequential type checker version!

In that case, we initialize a separate graph which reference is propagated upwards in the callstack. I'll point the relevant pieces of code in the TypeChecker module